### PR TITLE
make stock report helpers serializable

### PR DESCRIPTION
**Pull request by dannyroberts**
this gives timezone migration an entry point for saving what it is going to do
and rebuilding stock efficiently

### CODE REVIEW

File: corehq/apps/commtrack/sms.py [hunk 1]
```diff
@@ -370,7 +371,7 @@ def single_action_transactions(self, action, args):
                         case_id=self.case_id,
                         product_id=p.get_id,
                         action=const.StockActions.RECEIPTS,
-                        quantity=value.split('.')[1]
+                        quantity=Decimal(value.split('.')[1])
```
Comment (benrudolph) [inline corehq/apps/commtrack/sms.py]: was this just broken before?

**Comment by benrudolph** [comment]
test failure looks real, but digging the change.
```
======================================================================
ERROR: test_bad_code (custom.ewsghana.tests.handlers.stock_on_hand.StockOnHandTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/custom/ewsghana/tests/handlers/stock_on_hand.py", line 169, in test_bad_code
    self.run_script(a)
  File "/home/travis/build/dimagi/commcare-hq/custom/logistics/tests/test_script.py", line 47, in run_script
    self.assertEqual(msg.text, unicode(command['text']))
AttributeError: 'NoneType' object has no attribute 'text'
```
also love the name of the test that broke

File: corehq/apps/commtrack/sms.py (hunk 1 above)
Comment (dannyroberts) [inline corehq/apps/commtrack/sms.py]: No I think it just wasn't being validated upfront, and that django's DecimalField accepts a string without complaining.

**Comment by dannyroberts** [comment]
:white_check_mark:

**Pull request merged**